Tylko w submount-0.9/subfs-0.9: o
diff -urp submount-0.9./subfs-0.9/subfs.c submount-0.9/subfs-0.9/subfs.c
--- submount-0.9./subfs-0.9/subfs.c	2007-04-11 18:34:31.256507000 +0200
+++ submount-0.9/subfs-0.9/subfs.c	2007-04-11 20:01:07.999922000 +0200
@@ -17,7 +17,10 @@
 #include <asm/uaccess.h>
 #include <linux/list.h>
 #include <linux/mount.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,20)
 #include <linux/namespace.h>
+#endif
 #include <linux/namei.h>
 #include <linux/dcache.h>
 #include <linux/sysfs.h>
@@ -25,7 +28,6 @@
 #include <asm/signal.h>
 #include <linux/signal.h>
 #include <linux/sched.h>
-#include <linux/version.h>
 #include <linux/rcupdate.h>
 
 #include "subfs.h"
@@ -313,7 +315,9 @@ static struct inode *subfs_make_inode(st
 	if (ret) {
 		ret->i_mode = mode;
 		ret->i_uid = ret->i_gid = 0;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,20)
 		ret->i_blksize = PAGE_CACHE_SIZE;
+#endif
 		ret->i_blocks = 0;
 		ret->i_atime = ret->i_mtime = ret->i_ctime = CURRENT_TIME;
 		ret->i_fop = &subfs_file_ops;
Tylko w submount-0.9/subfs-0.9: subfs.c~
Tylko w submount-0.9/subfs-0.9: .subfs.o.d
Tylko w submount-0.9/subfs-0.9: .tmp_versions
